File: 0807-max-increase-to-keep-city-skyline/0807-max-increase-to-keep-city-skyline.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        
        row = []
        col = []
        for i in range(len(grid)):
            row.append(max(grid[i]))
        for i in range(len(grid[0])):
            maxi = 0
            for j in range(len(grid)):
                maxi = max(maxi,grid[j][i])
            col.append(maxi)
        sum = 0
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                sum += min(row[i],col[j])-grid[i][j]
                
        return sum
                
        

File: 0807-max-increase-to-keep-city-skyline/0807-max-increase-to-keep-city-skyline.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        vector<int> row;
        int n = grid.size();
        int m = grid[0].size();
        vector<int> col;
        for (int i=0;i<n;i++) {
            int maxi=0;
            for (int j=0;j<m;j++) {
                maxi = max(maxi,grid[i][j]);
                
            }
            row.push_back(maxi);
        }
        
        for (int i=0;i<m;i++) {
            int maxi=0;
            for (int j=0;j<n;j++) {
                maxi = max(maxi,grid[j][i]);
                
            }
            col.push_back(maxi);
        }
        
        int sum = 0;
        for (int i=0;i<n;i++) {
            
            for (int j=0;j<m;j++) {
                
                sum += min(row[i],col[j])-grid[i][j];
            
            }
        }
        
        return sum;
    }
};